simpleflow/execute.py

```diff
@@ -409,7 +411,25 @@ def main():
         metavar='N',
         help='error file descriptor',
     )
+    parser.add_argument(
+        '--kill-children',
+        action='store_const',
+        const=True,
+        default=False,
+        help='kill child processes on exit',
+    )
     cmd_arguments = parser.parse_args()
+
+    def kill_child_processes():
+        process = psutil.Process(os.getpid())
+        children = process.children(recursive=True)
+
+        for child in children:
```

Comment:
well, here's psutil:
```python
        def _send_signal(self, sig):
            assert not self.pid < 0, self.pid
            if self.pid == 0:
                # see "man 2 kill"
                raise ValueError(
                    "preventing sending signal to process with PID 0 as it "
                    "would affect every process in the process group of the "
                    "calling process (os.getpid()) instead of PID 0")
            try:
                os.kill(self.pid, sig)
            except OSError as err:
                if err.errno == errno.ESRCH:
                    if OPENBSD and pid_exists(self.pid):
                        # We do this because os.kill() lies in case of
                        # zombie processes.
                        raise ZombieProcess(self.pid, self._name, self._ppid)
                    else:
                        self._gone = True
                        raise NoSuchProcess(self.pid, self._name)
                if err.errno in (errno.EPERM, errno.EACCES):
                    raise AccessDenied(self.pid, self._name)
                raise
```